ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/jdbc/HiveJdbcImplementor.java

```diff
@@ -68,6 +68,18 @@ public HiveJdbcImplementor(SqlDialect dialect, JavaTypeFactory typeFactory) {
   @Override public Result visit(Sort e) {
     Result x = visitInput(e, 0, Clause.ORDER_BY, Clause.FETCH, Clause.OFFSET);
     Builder builder = x.builder(e);
+    
+    if (e.fetch != null) {
+      builder = x.builder(e);
+      builder.setFetch(builder.context.toSql(null, e.fetch));
+      x = builder.result();
+    }
+    if (e.offset != null) {
+      builder = x.builder(e);
+      builder.setOffset(builder.context.toSql(null, e.offset));
+      x = builder.result();
+    }
+    
```

Comment:
The first question that comes to mind is if this problem affects Calcite or if it is only Hive specific. I get the impression that it is mostly due to the Hive override that seems somewhat broken. From a quick look, I think we can get rid of this override completely and rely entirely on Calcite `visit(Sort)` method.

Reply:
Removed the overridden method from the class.